### ISSUE TITLE
Inconsistent loss values in "fit" and "evaluate" when "shuffle=False"

### ISSUE THREAD

**Issue by sadeghmir** [fchollet/keras#3326]
I'm training a single-layer LSTM with batch size 1, and the last epoch has the following output:

 `Epoch 1/1
22439/22439 [==============================] - 28s - loss: 902.9773`

Right after this epoch (training is done), I run `model.evaluate` **on the same data and labels** and get a completely different loss value. Here is the output:

`22418/22439 [============================>.] - ETA: 0s3694.0894752540389`

This happens only when I set `shuffle=False`. Otherwise the two loss values are very close. Why is this happening? Right after evaluating, I ran `fit` again for a single epoch and got loss close to 900 again.

**Comment by sadeghmir** [comment]
I implemented a similar LSTM network in Theano and achieved the same results as above! So this is no issue with Keras. I believe the reason that this inconsistency happens is that when `shuffle=False`, the model (while training) adapts its weights quickly to very recent examples and becomes more effective in predicting the current example (i.e. it finds a simpler solution to the problem and it will most likely not learn global regularities at all). In evaluation, the model does not have this chance of adapting to recent examples and will be lost...
